pkg/controller/jobs/jobset/jobset_webhook.go

```diff
@@ -57,9 +67,33 @@ func (w *JobSetWebhook) Default(ctx context.Context, obj runtime.Object) error {
 	log.V(5).Info("Applying defaults", "jobset", klog.KObj(jobSet))
 
 	jobframework.ApplyDefaultForSuspend(jobSet, w.manageJobsWithoutQueueName)
+
+	if canDefaultManagedBy(jobSet.Spec.ManagedBy) {
+		localQueueName, found := jobSet.Labels[constants.QueueLabel]
+		if !found {
+			return nil
+		}
```

Comment:
@mimowo @alculquicondor Does this mean that we don't support MultiKueue for the Job without queue name? 

https://github.com/kubernetes-sigs/kueue/blob/0a2c120c037fa9416736f87052fba6f60c4e27c6/apis/config/v1beta1/configuration_types.go#L41-L48

Reply:
If the user does not specify the `queueName` we cannot know if the target CQ is MK, so we don't default `managedBy`. Still, it could work if the user manually specified the `managedBy: multikueue`.

Reply:
Thank you for the clarifications!
So, expanding the documentation would be better.
@vladikkuzn Could you update this docs?

> {{% alert title="Note" color="note" %}} The same jobset-sample.yaml file from [single cluster environment](https://github.com/kubernetes-sigs/kueue/blob/8ba9d7765022d2f81f070d740ce4c2049d3f74ec/site/content/en/docs/tasks/run/jobsets.md#single-cluster-environment) can be used in a [MultiKueue environment](https://github.com/kubernetes-sigs/kueue/blob/8ba9d7765022d2f81f070d740ce4c2049d3f74ec/site/content/en/docs/tasks/run/jobsets.md#multikueue-environment). The spec.managedBy field will be set to kueue.x-k8s.io/multikueue automatically, if not specified. {{% /alert %}}

Reply:
Makes sense, maybe:
```
The same jobset-sample.yaml file from [single cluster environment] can be used in a
[MultiKueue environment].  The `spec.managedBy` field will be set to `kueue.x-k8s.io/multikueue`
automatically, if not specified, as long as  the `kueue.x-k8s.io/queue-name` annotation
is specified and the corresponding Cluster Queue uses the [Multi Kueue admission check]. 
```
I think we don't need to mention `manageJobsWithoutQueueName: true` specifically, because the field is defaulted regardless, it just cannot be changed. So we could add maybe another sentence: "Please also note that `spec.managedBy` field is immutable". 

WDYT @tenzen-y ?

Reply:
> I think we don't need to mention manageJobsWithoutQueueName: true specifically, because the field is defaulted regardless, it just cannot be changed. 

That makes sense. Instead of my recommendation above, how about expanding KueueConfig API documentation?

> "Please also note that spec.managedBy field is immutable"

This limitation doesn't come from kueue; it is brought us from JobSet, right?
So, it might be better to improve JobSet documentation and just put it link here. HDYT?

https://github.com/kubernetes-sigs/jobset/blob/7752a365fa5c9ab497c5b61d4dcd7c9e3609e9cb/pkg/webhooks/jobset_webhook.go#L231

Reply:
> That makes sense. Instead of my recommendation above, how about expanding KueueConfig API documentation?

sgtm, do you have a specific suggestion?

> This limitation doesn't come from kueue; it is brought us from JobSet, right?

correct

> So, it might be better to improve JobSet documentation and just put it link here. HDYT?

We could add a note [here](https://github.com/kubernetes-sigs/jobset/blob/7752a365fa5c9ab497c5b61d4dcd7c9e3609e9cb/api/jobset/v1alpha2/jobset_types.go#L97), but it seems there is no JobSet reference built for now: https://jobset.sigs.k8s.io/docs/reference/.

Reply:
> Moreover, when using the MultiKueue feature, users must specify the ".spec.managedBy" name in the Jobs by themselves.

Maybe, but there is no code in the webhook that depends on `manageJobsWithoutQueueName` currently. Also, the webhook is JobSet-specific at the moment. As long as the queueName annotation is supplied, then `managedBy` is defaulted for JobSet. Maybe the note around [here](https://github.com/kubernetes-sigs/kueue/blob/8ba9d7765022d2f81f070d740ce4c2049d3f74ec/site/content/en/docs/tasks/run/jobsets.md#example-jobset) is enough?

Reply:
> Maybe, but there is no code in the webhook that depends on manageJobsWithoutQueueName currently.

Yeah, you're right... It seems that my understanding for the `manageJobsWithoutQueueName`. So, my recommendation is no valid.

Reply:
> That makes sense. Regarding reference, creating an issue would be better in the JobSet side.

I have opened the Issue: https://github.com/kubernetes-sigs/jobset/issues/566
and the PR: https://github.com/kubernetes-sigs/jobset/pull/565

Reply:
Thank you!